rpc/handler.go

```diff
@@ -92,6 +92,24 @@ func newHandler(connCtx context.Context, conn jsonWriter, idgen func() ID, reg *
 	return h
 }
 
+func NewHandler(connCtx context.Context, reg *serviceRegistry) *handler {
```

Comment:
It's pretty confusing to have `newHandler` live next to `NewHandler`. Can you articulate the difference between these in the method name? Ideally we wouldn't need to create a whole new function, just tweak the interface to make it more generic.